### PR TITLE
[Encryption] Add datasources destructive reset

### PR DESCRIPTION
**Pull request by jfreden**
Datasource credentials are now wiped rather than the entire datasource entry being dropped during destructive reset. Non-credential config (name, type, description, plaintext settings) is preserved so users only need to re-enter credentials rather than recreate datasources from scratch. The API returns `null` for wiped credentials instead of `::es_redacted::` to make the broken state visible. A secret value can't be set to null through the API, only through the destructive path. 

The primary encryption key (PEK) is automatically generated and stored encrypted on disk. If you lose access to whatever protects that on-disk encrypted PEK, the disk fails, the hardware security module is unavailable, the key encryption material is gone, you simply can't decrypt it anymore, and therefore all secrets become permanently unreadable. `POST /_encryption/_reset` is the recovery path: you accept that the encrypted data is unrecoverable, destroy the inaccessible PEK, generate a fresh one, and use the `onDestructiveReset` callbacks to clean up the now-useless encrypted values in cluster state. With this change, datasource configs survive intact so users only need to re-enter credentials rather than rebuild everything from scratch.

How it works:

1. `POST /_encryption/_reset` is called with `accept_data_loss: true` - the explicit flag is there so no one triggers it accidentally.
2. The transport action destroys the PEK and, in the same atomic cluster state update, calls every registered `EncryptedDataHandler.onDestructiveReset` callback.
3. Each handler decides what to do with its encrypted data. The default (return `null`) drops the entire custom metadata (behaviour before this PR). Our new implementation instead returns a skeleton with secrets nulled out.
4. The result is written to cluster state in one atomic step - the old PEK and all encrypted values are gone together.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSetting.java [hunk 1]
```diff
@@ -109,9 +109,10 @@ public Object rawValue() {
         return value;
     }
 
-    /** Returns the masked sentinel for secrets, or the plaintext value otherwise. Safe for REST responses. */
+    /** Returns the masked sentinel for set secrets, {@code null} for wiped secrets, or the plaintext value for non-secrets. Safe for REST responses. */
     public Object presentationValue() {
-        return secret ? MASK_SENTINEL : value;
+        if (secret == false) return value;
+        return value == null ? null : MASK_SENTINEL;
```
Comment (quackaplop) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSetting.java:115]: You mentioned you weren't thrilled about returning `null` here and couldn't think of a better option — I think there is one: a distinct sentinel like `::es_reset_cleared::`, parallel to `::es_redacted::`. That keeps the wiped state visible and, importantly, separable from "optional secret never set" — which already reads as `null` here too, so right now the two are indistinguishable.

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSettingTests.java [hunk 2]
```diff
@@ -80,6 +80,11 @@ public void testPresentationValueSecret() {
         assertEquals("::es_redacted::", setting.presentationValue());
     }
 
+    public void testPresentationValueWipedSecret() {
```
Comment (quackaplop) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSettingTests.java:83]: Can you add a round-trip test for the wiped state? `(null, true)` is the one new persisted value this PR introduces, and since `DataSourceMetadata` is GATEWAY-persisted (SMILE on disk) it has to survive a restart — but the existing tests only round-trip `(null, false)`. It passes today; this just pins it so a later change to `presentationValue`/`writeTo` can't silently break a reset state's reload:

```java
public void testWipedSecretRoundTrip() throws IOException {
    var wiped = new DataSourceSetting(null, true);
    var wire = writeableRoundTrip(wiped);
    assertNull(wire.rawValue());
    assertTrue(wire.secret());
    assertXContentRoundTrip(wiped);
    assertSmileRoundTrip(wiped);
}
```

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/DataSourceEncryptedDataHandler.java [hunk 3]
```diff
@@ -32,6 +32,33 @@ public String customName() {
...
 
+    /**
+     * On destructive reset, wipe only the encrypted credential values while preserving the rest of
+     * each data source's configuration (name, type, description, non-secret settings). Users will
+     * see {@code null} for credentials in the API and need only re-provision those — not recreate
+     * the entire data source from scratch.
+     */
+    @Override
+    public DataSourceMetadata onDestructiveReset(DataSourceMetadata current) {
+        if (current == null || current.dataSources().isEmpty()) {
```
Comment (quackaplop) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/DataSourceEncryptedDataHandler.java:43]: Is the empty-but-present case meant to return non-null `current` here? The default `onDestructiveReset` contract drops the whole custom by returning `null`, so this diverges — fine if intended, just flagging.

**Review by quackaplop — COMMENTED** [review]
LGTM, a few comments before we ship

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/DataSourceEncryptedDataHandler.java (hunk 3 above)
Comment (jfreden) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/DataSourceEncryptedDataHandler.java:43]: Right, it's different from the default because we know a little more since we're in the context of the datasources handler. `current.dataSources().isEmpty()` just means that we can't have any secrets, hence nothing to reset.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSetting.java (hunk 1 above)
Comment (jfreden) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSetting.java:115]: Yes, it's tempting. I decided to stick with `null` because I already find `::es_redacted::` a little strange (even if that's the current preferred way in ES).

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSettingTests.java (hunk 2 above)
Comment (jfreden) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/datasources/metadata/DataSourceSettingTests.java:83]: Thanks! Will add.

**Review by quackaplop — APPROVED** [review]
